uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -451,28 +451,41 @@ public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDa
    *   And array of values from the Leave Request Status OptionGroup
    * @param bool $publicHolidaysOnly
    *   When true, will get the balance change only for the Public Holiday Leave Requests
+   * @param bool $expiredOnly
+   *   When true, will only count the balance change for expired days
    *
    * @return array
    */
   public static function getBalanceChangeByAbsenceType(
     $contactID,
     $periodID,
     $leaveRequestStatus = [],
-    $publicHolidaysOnly = false
+    $publicHolidaysOnly = false,
+    $expiredOnly = false
```

Comment:
To be honest, I think this whole method does not belong to the LeaveRequest BAO. If you take a look at its implementation, it doesn't use any of that BAO's methods and gets all of it's data from LeaveBalanceChange and LeavePeriodEntitlement. There are two places where I think it should probably be:

- On the `civicrm_api3_leave_request_getbalancechangebyabsencetype`() function (yeah, directly on the API function. I think this is ok because this is the only place the method will be used anyway and it also avoids duplication of tests between BAO and API)
- On a separate class (like an API service/class, for example). The problem would only the the duplication of tests between API and Service

About using `getLeaveRequestBalance()` for it. It's not exactly the same thing. That method is specific about Leave Requests and including the Brought Forward on it would even be more confusing. That is the reason I also added the param to the `LeaveBalanceChange.getBalanceForEntitlement()` method. The Brought Forward is part of the entitlement and we can think of accrued TOIL as being part of the entitlement as well, since, event though it's stored as a Leave Request internally, it add days to the entitlement.